Serializer/JsonApiDeserializationVisitor.php

```diff
@@ -60,11 +60,14 @@ public function visitProperty(PropertyMetadata $metadata, $data, Context $contex
             isset($data['relationships'][$propertyName])) { // TODO: add this property
             $included = isset($data['included']) ? $data['included'] : [];
 
+            $visit = false;
             $relationship = [];
-            if (isset($data['data']['relationships'][$propertyName]['data'])) {
+            if (array_key_exists('data', $data['data']['relationships'][$propertyName])) {
                 $relationship = $data['data']['relationships'][$propertyName]['data'];
-            } elseif (isset($data['relationships'][$propertyName]['data'])) {
+                $visit = true;
+            } elseif (array_key_exists('data', $data['relationships'][$propertyName])) {
                 $relationship = $data['relationships'][$propertyName]['data'];
+                $visit = true;
```

Comment:
@vlady777  you are able to cover this functionality with functional tests. Please see \Mango\Bundle\JsonApiBundle\Tests\Serializer\DeserializationTest

Reply:
Ticket was created for that. See related tickets.